### DISCUSSION TITLE
how do I use look-around?

### DISCUSSION THREAD

**Discussion by zhguokai** [rust-lang/regex#841] [Q&A] [answered]
let regex = regex::Regex::new(r"(?<=\#{)[^{}]+(?=})").unwrap();
let rule = "select ${abc} from #{b}";

 for item in regex.captures_iter(rule) {
      println!("{}", item.get(0).unwrap().as_str())
  }

report:

called `Result::unwrap()` on an `Err` value: Syntax(
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
regex parse error:
    (?<=\${)[^{}]+(?=})
    ^^^^
error: look-around, including look-ahead and look-behind, is not supported
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
)

why

**Comment by zhguokai** [comment]
how to fix

**Comment by BurntSushi** [comment, marked as answer]
The first sentences of the docs for this crate address and explain why:

> This crate provides a library for parsing, compiling, and executing regular expressions. Its syntax is similar to Perl-style regular expressions, but lacks a few features like look around and backreferences. In exchange, all searches execute in linear time with respect to the size of the regular expression and search text.

**Reply by zhguokai** [reply]
thanks, but how to fix it? change express like Perl ? or not support

**Reply by BurntSushi** [reply]
I think the docs are pretty clear. The feature you want isn't supported by this crate. You either need to stop using look-around, or use a different regex engine.

**Reply by zhguokai** [reply]
thanks, it is difficult for me to under the doc write in English,  which regex engine can I use for look-around?

**Reply by zhguokai** [reply]
i will try  "fancy-regex" ?

**Reply by BurntSushi** [reply]
Sure, go ahead. That's not my area of expertise.

**Comment by zhguokai** [comment]
fancy_regex
